04-Lab/FunctionalSpatialImageProcessing.ipynb

```diff
@@ -0,0 +1,517 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "# Copy files from previous labs (or download) so it is easier work with them here\n",
+    "\n",
+    "# Files from 03 Lab that include our transforms for T1->MNI\n",
+    "# DECIDE WHETHER YOU WANT TO COPY (cp) OR DOWNLOAD THE FILES NECESSARY FOR THIS LAB\n",
+    "# default\n",
+    "ans=\"dl\"\n",
+    "\n",
+    "if [ \"${ans}\" == \"cp\" ]; then\n",
+    "    cp -t ./ \\\n",
+    "       ../03-Lab/my_T1_brain.nii.gz \\\n",
+    "       ../03-Lab/my_T1.nii.gz \\\n",
+    "       ../03-Lab/MNI152_T1_2mm_brain.nii.gz \\\n",
+    "       ../03-Lab/MNI152_T1_2mm.nii.gz \\\n",
+    "       ../03-Lab/T1_2_MNI.mat \\\n",
+    "       ../03-Lab/T1_2_MNI_warp.nii.gz\n",
+    "    cp ../01-Lab/osfshare/sub-97_task-flanker_bold.nii.gz my_bold.nii.gz # Copy/rename functional image from osfshare\n",
+    "elif [ \"${ans}\" == \"dl\" ]; then\n",
+    "    wget --quiet -O 04-Lab_data.tar.gz https://osf.io/4yz8m/download &&\\\n",
+    "    tar -xf 04-Lab_data.tar.gz --strip-components=2 &&\\\n",
+    "    rm 04-Lab_data.tar.gz\n",
+    "else\n",
+    "    printf 'YOU FORGOT TO CHANGE ANS TO \"cp\" OR \"dl\"'\n",
+    "fi\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Registration of functional image to T1\n",
+    "\n",
+    "Summary of steps \n",
+    "* Extract reference volume from the 4D functional image data set \n",
+    "    * fslroi tool is handy for this\n",
+    "   \n",
+    "* Estimation: \n",
+    "    * Boundary Based Registration (BBR) which in theory is most accurate but takes a bit of time to run (try at home)\n",
+    "    * 6 degrees of freedom (DOF) co-registration (faster, practice in class)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# There are 315 volumes in this functional data set, it doesn't make a big deal whether you round up or down\n",
+    "315/2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "fslroi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Example usage to name our middle volume as example_func\n",
+    "# Reads: take input of my_bold, give back example func after extracting volumes starting at vol 157 with length 1\n",
+    "# Remember since indexing of volumes starts at zero, this is the 158th volume\n",
+    "fslroi my_bold.nii.gz example_func.nii.gz 157 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# skull strip the reference epi so that skull information doesn't influence the registration to the T1w brain.\n",
```

Comment:
I do skull stripping on the bold before registration to the T1w brain, that appears to be how it's done in fmriprep. can roll this change back if there's a reason keep the skull information.